CMakeLists.txt

```diff
@@ -386,11 +390,13 @@ endif()
 
 add_subdirectory(res)
 add_subdirectory(src)
-add_subdirectory(exposed)
+if(ENABLE_DEMO)
+    add_subdirectory(exposed)
+endif()
 if(ENABLE_TESTS)
     add_subdirectory(test)
 endif()
-if(CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+if(ENABLE_BENCH)
```

Comment:
Maybe. But maybe keep the original conditions too do we want someone running benchmarks in debug mode and complaining?

Reply:
We could do something like:

```cmake
if(ENABLE_BENCH)
  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
    message(FATAL_ERROR "Benchmarks cannot be enabled in Debug mode, please set ENABLE_BENCH=OFF or use another CMAKE_BUILD_TYPE value")
  endif()
  ...
```
?

Reply:
Thinking about it more I guess a warning makes more sense:

```cmake
if(ENABLE_BENCH)
    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
        message(WARNING "Benchmarks were automatically disabled because you are building in Debug mode, please use another CMAKE_BUILD_TYPE to enable benchmarks. disabling benchmarks...")
    endif()
    ...
 ```